src/compile/legend/parse.ts

```diff
@@ -111,13 +113,122 @@ export function parseLegendForChannel(model: UnitModel, channel: NonPositionScal
+  const selections: InteractiveSelections[] = [];
+  // Look over all selections
+  forEachSelection(model, selCmpt => {
+    if (selCmpt['fields']) {
+      selections.push({name: selCmpt.name, store: stringValue(selCmpt.name + STORE), fields: selCmpt['fields']});
+    }
+  });
+
+  // Quit if no selections have projections
+  if (!selections.length) {
+    return [];
+  }
+
+  // Encoding channels should fully populate selections
+  let selectionFields: string[] = [].concat.apply([], selections.map(s => s.fields)); // Flatten array
+  selectionFields = selectionFields.filter((v, i, a) => a.indexOf(v) === i); // Get unique elements
+
+  let encodingFields: string[] = [];
+  [COLOR, OPACITY, SIZE, SHAPE].forEach(channel => {
+    const fieldDef = model.fieldDef(channel);
+    if (
+      fieldDef &&
+      !(fieldDef.hasOwnProperty('bin') || fieldDef.hasOwnProperty('aggregate') || fieldDef.hasOwnProperty('timeUnit'))
+    ) {
+      encodingFields.push(fieldDef.field);
+    }
+  });
+
+  encodingFields = encodingFields.filter((v, i, a) => a.indexOf(v) === i); // Get unique elements
+  const differenceFields = selectionFields.filter(x => encodingFields.indexOf(x) === -1);
+  if (differenceFields.length) {
+    return [];
+  }
+  return selections;
+}
+
+function updateInteractiveLegendComponent(
+  model: UnitModel,
+  legendEncode: LegendEncode,
+  channel: NonPositionScaleChannel,
+  interactiveSelections: InteractiveSelections[]
+): LegendEncode {
+  switch (channel) {
+    case COLOR:
+    case OPACITY:
+    case SIZE:
+    case SHAPE:
+      break;
+    default:
+      return legendEncode;
+  }
+  const field = model.fieldDef(channel).field;
+
+  // Choose the selection with highest specifictiy of projection containing the field
+  let selectionIndex: number;
+  let maxFields = 0;
+  interactiveSelections.forEach((s, i) => {
+    if (s.fields.length > maxFields && s.fields.indexOf(field) > -1) {
+      maxFields = s.fields.length;
+      selectionIndex = i;
+    }
+  });
+  if (!maxFields) {
+    return legendEncode;
+  }
+  const maxProjSelection = interactiveSelections[selectionIndex];
+
+  const updatedLegendEncode = legendEncode;
+  let updateValue;
+  ['labels', 'symbols'].forEach(part => {
+    if (updatedLegendEncode.hasOwnProperty(part)) {
+      updateValue = updatedLegendEncode[part].update;
+    } else {
+      updateValue = {opacity: {value: 0.7}};
+    }
+
+    let test = `!(length(data(${maxProjSelection.store}))) || ${VL_SELECTION_TEST}(${
+      maxProjSelection.store
+    }, {${field}: datum.value})`;
+    if (maxProjSelection.fields.length > 1) {
+      test = `!${maxProjSelection.name}_${field}_legend || datum.value === ${maxProjSelection.name}_${field}_legend`;
+    }
+    if (part === 'symbols' && channel === OPACITY) {
+      let strokeValue = '#000000';
+      if (updateValue.stroke) {
+        strokeValue = updateValue.stroke.value;
+      }
+      updateValue.stroke = [{test, value: strokeValue}, {value: '#aaaaaa'}];
```

Comment:
How does this work on a black background? I think we need to use a config option for these.

Reply:
I was planning to remove these hard-coded values. @domoritz what would be the best possible place in the spec to have config options such as color, opacity values or disabling interactive legends. Should this be incorporated as an optional field for every mark encoding channel?

Reply:
We should use the Vega-Lite config for this. It might make sense to have a legend property as well that allows users to override the value from the config.